near-sdk-sim/src/outcome.rs

```diff
@@ -0,0 +1,253 @@
+use crate::hash::CryptoHash;
+use crate::runtime::{init_runtime, RuntimeStandalone};
+use crate::transaction::{ExecutionOutcome, ExecutionStatus};
+use core::fmt;
+use near_primitives::transaction::ExecutionStatus::{SuccessReceiptId, SuccessValue};
+use near_primitives::types::AccountId;
+use near_sdk::borsh::BorshDeserialize;
+use near_sdk::serde::de::DeserializeOwned;
+use near_sdk::serde::export::Formatter;
+use near_sdk::serde_json::Value;
+use near_sdk::{serde_json, Gas};
+use std::borrow::Borrow;
+use std::cell::RefCell;
+use std::fmt::Debug;
+use std::io;
+use std::io::{Error, ErrorKind};
+use std::rc::Rc;
+
+pub type TxResult = Result<ExecutionOutcome, ExecutionOutcome>;
+
+/// An ExecutionResult is created by a UserAccount submitting a transaction.
+/// It wraps an ExecutionOutcome which is the same object returned from an RPC call.
+#[derive(Clone)]
+pub struct ExecutionResult {
+    runtime: Rc<RefCell<RuntimeStandalone>>,
+    outcome: ExecutionOutcome,
+}
+
+impl Debug for ExecutionResult {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        f.debug_struct("ExecutionResult").field("outcome", &self.outcome).finish()
+    }
+}
+
+impl Default for ExecutionResult {
+    fn default() -> Self {
+        ExecutionResult::new(
+            ExecutionOutcome::default(),
+            &Rc::new(RefCell::new(init_runtime(None).0)),
+        )
+    }
+}
+
+impl ExecutionResult {
+    #[doc(hidden)]
+    pub fn new(outcome: ExecutionOutcome, runtime: &Rc<RefCell<RuntimeStandalone>>) -> Self {
+        Self { runtime: Rc::clone(runtime), outcome }
+    }
+
+    /// Interpret the SuccessValue as a JSON value
+    pub fn get_json_value(&self) -> serde_json::Result<Value> {
```

Comment:
@evgenykuzyakov Should these methods also unwrap the value instead of returning a result?

Reply:
yep